### PR TITLE
Add rbac and psp

### PR DESCRIPTION
**Pull request by pipo02mix**
Towards [#2296](https://github.com/giantswarm/giantswarm/issues/2296)

Add RBAC based on the k8s API methods found on the project

### CODE REVIEW

File: helloworld-manifest.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: helloworld
```
Comment (kopiczko) [inline helloworld-manifest.yaml]: helloworld?

File: service/resource/legacyv2/resource.go [hunk 2]
```diff
@@ -145,6 +145,39 @@ func (r *Resource) newCreateChange(ctx context.Context, obj, currentState, desir
...
+	// Bind the service account with the cluster role of flannel operator pod security policy
+	{
+		clusterRoleBindingPodSecurityPolicy := newClusterRoleBindingPodSecurityPolicy(customObject)
+		_, err := r.k8sClient.RbacV1beta1().ClusterRoleBindings().Create(clusterRoleBindingPodSecurityPolicy)
+		if apierrors.IsAlreadyExists(err) {
+			r.logger.Log("debug", "clusterRoleBindingPodSecurityPolicy "+clusterRoleBindingPodSecurityPolicy.Name+" already exists", "event", "add", "cluster", customObject.Spec.Cluster.ID)
+		} else if err != nil {
+			return nil, microerror.Maskf(err, "creating clusterRoleBindingPodSecurityPolicy %s", clusterRoleBindingPodSecurityPolicy.Name)
+		}
+	}
```
Comment (kopiczko) [inline service/resource/legacyv2/resource.go]: Shouldn't those guys also be deleted somewhere?

File: helloworld-manifest.yaml (hunk 1 above)
Comment (pipo02mix) [reply helloworld-manifest.yaml]: WTF this file shhould not be there 🤒

File: service/resource/legacyv2/resource.go (hunk 2 above)
Comment (pipo02mix) [reply service/resource/legacyv2/resource.go]: indeed good point

File: service/resource/legacyv2/cluster_role_binding.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+		},
+		ObjectMeta: apismeta.ObjectMeta{
+			Name: networkNamespace(customObject.Spec) + "-psp",
+			Annotations: map[string]string{
+				VersionBundleVersionAnnotation: keyv2.VersionBundleVersion(customObject),
+			},
+			Labels: map[string]string{
+				"app":      app,
+				"cluster":  clusterName(customObject.Spec),
+				"customer": clusterCustomer(customObject.Spec),
```
Comment (kopiczko) [inline service/resource/legacyv2/cluster_role_binding.go]: https://github.com/giantswarm/fmt/blob/master/kubernetes/annotations_and_labels.md

This should be `giantswarm.io/cluster-id` and `giantswarm.io/customer-id`

Comment (kopiczko) [reply service/resource/legacyv2/cluster_role_binding.go]: I know it was maybe different, but it's something we should converge to.

**Comment by pipo02mix** [comment]
I leave this PR only for testing purposes but I split my work into multiple ones (for better understanding)

**Comment by pipo02mix** [comment]
Close this one in favor of the smaller PRs

**Pull request closed without merging**